### PR TITLE
[release 1.30] upgrade kubernetes version and docs

### PR DESCRIPTION
**Pull request by a7i**
closes [#1398](https://github.com/kubernetes-sigs/descheduler/issues/1398) and [#1399](https://github.com/kubernetes-sigs/descheduler/issues/1399)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module sigs.k8s.io/descheduler
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.1
```
Comment (jklaw90) [inline go.mod:5]: should it be `1.22.2` to match the dockerfile??

Comment (a7i) [reply go.mod:5]: yeah I should have taken this out. it kept complaning about `go mod tidy` if I didn't

Comment (a7i) [reply go.mod:5]: let me know if it looks correct now

**Comment by ingvagabund** [comment]
Let's update https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/descheduler/descheduler-presubmits-master.yaml to deploy k8s 1.30 before merging this. Even though the master HEAD is currently v0.30.0-alpha.3.

**Comment by a7i** [comment]
@ingvagabund we typically have done that after the release branch is created, see 1.29 PR: https://github.com/kubernetes/test-infra/pull/31497

**Comment by a7i** [comment]
@ingvagabund https://github.com/kubernetes/test-infra/pull/32633

**Pull request merged**